includes/login-register.php

```diff
@@ -159,6 +159,70 @@ function edd_log_user_in( $user_id, $user_login, $user_pass, $remember = false )
 
 }
 
+add_filter( 'wp_login_errors', 'edd_login_register_error_message', 10, 2 );
+/**
+ * Changes the WordPress login confirmation message when using EDD's reset password link.
+ *
+ * @since 2.10
+ * @param object \WP_Error $errors
+ * @param string $redirect
+ * @return void
+ */
+function edd_login_register_error_message( $errors, $redirect ) {
+	$action_is_confirm   = ! empty( $_GET['checkemail'] ) && 'confirm' === sanitize_text_field( $_GET['checkemail'] );
+	$edd_action_is_reset = ! empty( $_GET['edd_reset_password'] ) && 'confirm' === sanitize_text_field( $_GET['checkemail'] );
+	$redirect_url        = ! empty( $_GET['edd_redirect'] ) ? $_GET['edd_redirect'] : false;
+	if ( $action_is_confirm && $edd_action_is_reset && $redirect_url ) {
+		$errors->remove( 'confirm' );
+		$errors->add(
+			'confirm',
+			apply_filters(
+				'edd_login_register_error_message',
+				sprintf(
+					/* translators: %s: Link to the referring page. */
+					__( 'Follow the instructions in the confirmation email you just received, then <a href="%s">return to what you were doing</a>.', 'easy-digital-downloads' ),
+					esc_url( $redirect_url )
+				),
+				$redirect_url
+			),
+			'message'
+		);
+	}
+
+	return $errors;
+}
+
+/**
+ * Gets the lost password URL, customized for EDD. Using this allows the password
+ * reset form to redirect to the login screen with the EDD custom confirmation message.
+ *
+ * @since 2.10
+ * @return string
+ */
+function edd_get_lostpassword_url() {
+	$url = false;
+	// `wp_get_raw_referer` was added in WP 4.5; this check is required as we still support 4.4.
+	if ( function_exists( 'wp_get_raw_referer' ) ) {
+		$url = wp_get_raw_referer();
+	} else {
+		if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
+			$url = wp_unslash( $_REQUEST['_wp_http_referer'] );
+		} elseif ( ! empty( $_SERVER['HTTP_REFERER'] ) ) {
+			$url = wp_unslash( $_SERVER['HTTP_REFERER'] );
+		}
+	}
+	$url      = wp_validate_redirect( $url, edd_get_checkout_uri() );
+	$redirect = add_query_arg(
+		array(
+			'checkemail'         => 'confirm',
+			'edd_reset_password' => 'confirm',
+			'edd_redirect'       => esc_url_raw( $url ),
```

Comment:
This should probably be `urlencode()` instead of `esc_url_raw()`, then when we use it in `edd_login_register_error_message`, we should `urldecode()` it.